marklogic-data-hub-central/src/main/java/com/marklogic/hub/central/entities/search/EntitySearchManager.java

```diff
@@ -246,16 +247,16 @@ public void exportRows(JsonNode queryDocument, Long limit, OutputStream out) {
         List<SearchQuery.SortOrder> sortOrder = searchQuery.getSortOrder().orElse(new ArrayList<>());
         ArrayNode sortOrderNode = sortOrderToArrayNode(sortOrder);
 
-
-        JsonNode opticPlanNode = EntitySearchService.on(finalDatabaseClient).getOpticPlan(structuredQuery, searchText, queryOptions, entityTypeId, entityTypeId, limit, sortOrderNode, columns.stream());
-        StringHandle stringHandle = new StringHandle(opticPlanNode.toString());
-        RowManager rowManager = finalDatabaseClient.newRowManager();
-        try (ReaderHandle readerHandle = new ReaderHandle()) {
-            rowManager.resultDoc(rowManager.newRawPlanDefinition(stringHandle), readerHandle.withMimetype(CSV_CONTENT_TYPE));
+        EntitySearchService entitySearchService = EntitySearchService.on(finalDatabaseClient);
+        // Exporting directly from Data Service to avoid bug https://bugtrack.marklogic.com/55338 related to namespaced path range indexes
+        Reader export = entitySearchService.exportSearchAsCSV(structuredQuery, searchText, queryOptions, entityTypeId, entityTypeId, limit, sortOrderNode, columns.stream());
+        try (ReaderHandle readerHandle = new ReaderHandle(export).withMimetype(CSV_CONTENT_TYPE)) {
             readerHandle.write(out);
         } catch (IOException e) {
+            IOUtils.closeQuietly(out);
```

Comment:
We don't need this here since the "finally" block will always close the resource.